File: task4/restapi/src/main/java/model/Shop.java
package model;

import java.io.Serializable;

public class Shop implements Serializable {
    private long id;
    private double latitude;
    private double longitude;
    private String category;
    private String shopname;
    private String openingHours;
    private String website;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public double getLatitude() {
        return latitude;
    }

    public void setLatitude(double latitude) {
        this.latitude = latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public void setLongitude(double longitude) {
        this.longitude = longitude;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public String getShopname() {
        return shopname;
    }

    public void setShopname(String shopname) {
        this.shopname = shopname;
    }

    public String getOpeningHours() {
        return openingHours;
    }

    public void setOpeningHours(String openingHours) {
        this.openingHours = openingHours;
    }

    public String getWebsite() {
        return website;
    }

    public void setWebsite(String website) {
        this.website = website;
    }

    @Override
    public String toString() {
        String shopString = "name: " + shopname;
        shopString += (category != null && !category.equals("")) ? (", category: " + category) : "";
        shopString += (openingHours != null && !openingHours.equals("")) ? (", " + openingHours) : "";
        shopString += (website != null && !website.equals("")) ? (", " + website) : "";
        return shopString;
        }
}

File: task4/webview/src/app/classes/shop.ts
export class Shop {
    id: number;
    latitude: number;
    longitude: number;
    category: string;
    shopname: string;
    openingHours: string;
    website: string;

    constructor (id: number, latitude: number, longitude: number, category: string, shopname: string, openingHours: string, website: string) {
        this.id = id;
        this.latitude = latitude;
        this.longitude = longitude;
        this.category = category;
        this.shopname = shopname;
        this.openingHours = openingHours;
        this.website = website;
    }

    public static fromJson(json: Object) {
        return new Shop(json['id'], json['latitude'], json['longitude'], json['category'], json['shopname'], json['openingHours'], json['website']);
    }

    public generatePopupHTML() {
        var popup: string = "<b>" + this.shopname + "</b>";
        if (this.category != null && this.category != "") {
            popup += "<br>(" + this.category + ")";
        }
        if (this.openingHours != null && this.openingHours != "") {
            popup += "<br>" + this.openingHours;
        }
        if (this.website != null && this.website != "") {
            popup += "<br>" +  "<a href='" + this.website + "'>" + this.website + "</a>";
        }
        return popup;
    }
}